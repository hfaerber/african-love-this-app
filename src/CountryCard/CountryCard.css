.country-card {
  border: 1px solid pink;
  text-align: center;
  padding: 5px;
  width: 30%;
}

.country-name {
  font-size: 1em;
  font-weight: bold;
}

.country-capital {
  font-size: .9em;
  font-weight: bold;
}

.country-region {
  font-size: .7em;
  font-style: italic;
  color: lightgrey;
}

.empty-checkbox {
  width: 25px;
  /* display:none; */
}
