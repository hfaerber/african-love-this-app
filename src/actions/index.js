export const updateUser = user => ({
  type: 'UPDATE_USER',
  user
});

export const updateCountries = countries => ({
  type: 'UPDATE_COUNTRIES',
  countries
});

export const toggleStudyStatus = name => ({
  type: 'TOGGLE_STUDY_STATUS',
  name
});

export const updateSelectedFilter = filter => ({
  type: 'UPDATE_SELECTED_FILTER',
  filter
});

export const updateSearchQuery = query => ({
  type: 'UPDATE_SEARCH_QUERY',
  query
});

export const updateError = errorMessage => ({
  type: 'UPDATE_ERROR',
  errorMessage
});

export const updateLoadingStatus = status => ({
  type: 'UPDATE_LOADING_STATUS',
  status
});
