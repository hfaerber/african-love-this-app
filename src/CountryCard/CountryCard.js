import React from 'react';
import './CountryCard.css';
import emptybox from '../images/check-box-empty.svg';
import { toggleStudyStatus } from '../actions';
import { connect } from 'react-redux';


export const CountryCard = ({ name, capital, region, needsStudied, toggleStudyStatus }) => {
  let buttonText = needsStudied ? 'Got It!' : 'Still Learning';
  return(
    <section className='country-card'>
      <p className='country-name'>{name}</p>
      <p className='country-capital'>{capital}</p>
      <p className='country-region'>{region}</p>
      <button className='button-study-status' onClick={() => toggleStudyStatus(name)}
      >{buttonText}</button>
    </section>
  )
}

export const mapDispatchToProps = dispatch => ({
  toggleStudyStatus: name => dispatch(toggleStudyStatus(name))
})

export default connect(null, mapDispatchToProps)(CountryCard);
// <button>Still Learning</button>
// <img src={emptybox} className='empty-checkbox' alt='empty checkbox' />
// User clicks on got it
// 1 )style changes to fade that card out
// 2)  update status in the store for that country element
// Need action creator for "Toggle_studyStatus"
