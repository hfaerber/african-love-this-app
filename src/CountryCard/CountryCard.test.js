import React from 'react';
import { CountryCard, mapDispatchToProps } from './CountryCard';
import { toggleStudyStatus } from '../actions'
import { shallow } from 'enzyme';

describe('CountryCard', () => {
  let mockName;
  describe('CountryCard component', () => {
    let wrapper, mockToggleStudyStatus;
    beforeEach(() => {
      mockName = 'Benin';
      mockToggleStudyStatus = jest.fn();
      wrapper = shallow(<CountryCard
        name={mockName}
        capital='Capital of Benin'
        region='Region of Benin'
        needsStudied={true}
        toggleStudyStatus={mockToggleStudyStatus}
        />)
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })

    it('should match snapshot when needsStudied is false', () => {
      wrapper = shallow(<CountryCard
        needsStudied={false}
        />);
      expect(wrapper).toMatchSnapshot();
    })

    it('should invoke toggleStudyStatus prop on click', () => {
      wrapper.find('.button-study-status').simulate('click');
      expect(mockToggleStudyStatus).toHaveBeenCalledWith(mockName);
    })
  });

  describe('mapDispatchToProps', () => {
    mockName = 'Benin';
    it('should call dispatch with a toggleStudyStatus action when toggleStudyStatus is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = toggleStudyStatus('Benin');
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.toggleStudyStatus('Benin');
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  });
})
