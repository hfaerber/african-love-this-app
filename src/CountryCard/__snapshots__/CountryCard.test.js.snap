// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryCard CountryCard component should match snapshot 1`] = `
<section
  className="country-card"
>
  <p
    className="country-name"
  >
    Benin
  </p>
  <p
    className="country-capital"
  >
    Capital of Benin
  </p>
  <p
    className="country-region"
  >
    Region of Benin
  </p>
  <div
    className="div-status"
  >
    <button
      className="button-study-status"
      onClick={[Function]}
    >
      Got It!
    </button>
  </div>
</section>
`;

exports[`CountryCard CountryCard component should match snapshot when needsStudied is false 1`] = `
<section
  className="country-card"
>
  <p
    className="country-name"
  />
  <p
    className="country-capital"
  />
  <p
    className="country-region"
  />
  <div
    className="div-status"
  >
    <button
      className="button-study-status"
      onClick={[Function]}
    >
      Still Learning
    </button>
    <img
      alt="checked checkbox"
      className="img-checkbox"
      src="check-box-checked.svg"
    />
  </div>
</section>
`;
